tests.py

```diff
@@ -17,30 +17,19 @@
 matplotlib.use('agg')
 
 import nose
-from nose.plugins import attrib
-from matplotlib.testing.noseclasses import KnownFailure
 from matplotlib import default_test_modules
 
-plugins = [KnownFailure, attrib.Plugin]
-
-# Nose doesn't automatically instantiate all of the plugins in the
-# child processes, so we have to provide the multiprocess plugin with
-# a list.
-from nose.plugins import multiprocess
-multiprocess._instantiate_plugins = plugins
-
 
 def run(extra_args):
-    try:
-        import faulthandler
-    except ImportError:
-        pass
-    else:
-        faulthandler.enable()
+    from nose.plugins import multiprocess
+
+    matplotlib._init_tests()
 
-    if not os.path.isdir(
```

Comment:
I keep wanting to write a blog post about all the ways that diffs can confuse humans, but I fear it would be used by the NSA as a crib sheet for how to sneak malicious code into open source projects.